plugin/src/test/java/io/jenkins/plugins/casc/SchemaGenerationTest.java

```diff
@@ -0,0 +1,144 @@
+         * Used to generate the schema for root configurators
+         */
+        LinkedHashSet linkedHashSet = new LinkedHashSet<>(ConfigurationAsCode.get().getRootConfigurators());
+        Iterator<RootElementConfigurator> i = linkedHashSet.iterator();
+
+        StringBuilder schemaString = new StringBuilder();
+        schemaString.append("{\n" +
+                "  \"$schema\": \"http://json-schema.org/draft-06/schema#\",\n" +
+                "  \"id\": \"http://jenkins.io/configuration-as-code#\",\n" +
+                "  \"description\": \"Jenkins Configuration as Code\",\n" +
+                "  \"type\": \"object\",\n" +
+                "  \"properties\": {\n" +
+                "");
+
+        while (i.hasNext()) {
+
+            List<Attribute> attributeList = new ArrayList<>(i.next().getAttributes());
+            for (Attribute a : attributeList) {
+                String type;
+                if (a.type.toString().equals("int")) {
+                    type = "int";
+                } else if (a.type.toString().equals("class java.lang.String")) {
+                    type = "string";
+                } else if (a.type.toString().equals("boolean")) {
+                    type = "boolean";
+                } else {
+                    type = "object";
+                }
+                schemaString.append(a.name + ": " + "{\n" +
+                        "      \"type\": \"" + type + "\",\n" +
+                        "      \"$ref\": \"#/definitions/" + a.type.getName() + "\"\n" +
+                        "    },");
+            }
+        }
+        System.out.println(schemaString);
+
+        /**
+         * Used to generate the schema for the descriptors
+         * Finds out the instance of each of the configurators
+         * and gets the required descriptors from the instance method.
+         * Appending the oneOf tag to the schema.
+         */
+
+        schemaString.append("schema\" : {\n" +
+                "    \"oneOf\": [");
+
+        ConfigurationAsCode configurationAsCode = ConfigurationAsCode.get();
+        configurationAsCode.getConfigurators()
+                .stream()
+                .forEach(configurator -> {
+
+                    DescriptorExtensionList descriptorExtensionList = null;
+                    System.out.println(configurator.getClass().getName());
+                    if(configurator instanceof HeteroDescribableConfigurator) {
+                        HeteroDescribableConfigurator heteroDescribableConfigurator = (HeteroDescribableConfigurator) configurator;
+                        descriptorExtensionList = j.jenkins.getDescriptorList(heteroDescribableConfigurator.getTarget());
+
+                    } else if (configurator instanceof DataBoundConfigurator){
+                        DataBoundConfigurator dataBoundConfigurator = (DataBoundConfigurator) configurator;
+                        descriptorExtensionList = j.jenkins.getDescriptorList(dataBoundConfigurator.getTarget());
```

Comment:
@casz What about this line where I need to call  `j.jenkins` because I do not have the rule defined in the class.

Reply:
`Jenkins.get()` instead